Uniswap-Swap-Event-Detection-Bot/src/agent.ts

```diff
@@ -0,0 +1,88 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType, getJsonRpcUrl } from "forta-agent";
+
+import { ethers } from "ethers";
+import LRU from "lru-cache";
+import { sharedConstant, sharedConstantType } from "./shared";
+
+function computePoolAddress(
+  token0: string,
+  token1: string,
+  deployer: string,
+  initCodeHash: string,
+  fee: string
+): string {
+  const salt: string = ethers.utils.keccak256(
+    ethers.utils.defaultAbiCoder.encode(["address", "address", "uint24"], [token0, token1, fee])
+  );
+  const computedPoolAddress: string = ethers.utils.getCreate2Address(deployer, salt, initCodeHash);
+  return computedPoolAddress;
+}
+
+const dataCache: LRU<string, string[]> = new LRU<string, string[]>({ max: 500 });
```

Comment:
```suggestion
const dataCache: LRU<string, string[]> = new LRU<string, string[]>({ max: 10_000 });
```
(Typically we use this size)